content/tutorial-ios/ios-06.md

```diff
@@ -0,0 +1,193 @@
+# Tutorial 06 - More Mutations
+
+Welcome to the 6th exercise in the **iOS Track** of this Apollo Client Tutorial!
+
+
+## Goal
+
+The **goal** of this exercise is to implement features for updating and deleting existing Pokemons from the Pokedex. This will give you more familiarity with using mutations.
+
+
+## Introduction
+
+Open the directory that contains the 6th exercise (`exercise-06`) and open `pokedex-apollo.xcworkspace`. It already contains a running version of the code you wrote in the previous lesson. Note that we added some functionality to the `PokemonDetailViewController` that allows us to switch to an _editing state_ in which we can change the name and image URL using the corresponding text fields. However, any changes that we make right now will have no effect - that's what we want to implement in this exercise.
+
+
+## Updating An Existing Pokemon
+
+### Defining An `UpdatePokemonMutation` 
+
+As a first step, we want to implement update functionality for our Pokemons. This will be done through a mutation that we need to define upfront. Since the mutation will only be used inside the `PokemonDetailViewController`, now is a good time to create a new `.graphql` file that contains all queries/mutations for the `PokemonDetailViewController`. As before, create a new _empty_ file in Xcode and call it `PokemonDetailViewController.graphql`. Then, copy the following mutation into it:
+
+```graphql
+mutation UpdatePokemon($id: ID!, $name: String!, $url: String!) {
+  updatePokemon(id: $id, name: $name, url: $url) {
+    ... PokemonDetails
+  }
+}
+```
+
+Note that if we hadn't made `$name` and `$url` mandatory, the user of this mutation might not provide a value for either of them which would delete the value for the field with a missing parameter. So, here we require both values to prevent the user from accidentally deleting them.
```

Comment:
I am not sure I understand this passage correctly. To be clear, not including the url argument for instance doesn't  delete the current url. only when you provide another  url value will the current URL be overwritten.